examples/src/main/java/kafka/examples/Consumer.java

```diff
@@ -91,9 +99,13 @@ public void run() {
                     // we can't recover from these exceptions
                     Utils.printErr(e.getMessage());
                     shutdown();
+                } catch (OffsetOutOfRangeException | NoOffsetForPartitionException e) {
+                    // invalid or no offset found without auto.reset.policy
+                    Utils.printOut("Invalid or no offset found, using latest");
+                    consumer.seekToEnd(emptyList());
+                    consumer.commitSync();
```

Comment:
Is this commit necessary?

Reply:
In the exactly-once demo (auto commit disabled), what happens if you seek to the end and in the next cycles there are no transactions to process? I think you will seek again after every consumer restart, until some transaction is processed and its offsets are committed. I know this can't happen in this demo, but could happen in theory, so I think this commit is correct.

Reply:
Fair enough.